crates/chain/tests/multi_node/mempool_tests.rs

```diff
@@ -1627,3 +1627,99 @@ async fn heavy_evm_mempool_fork_recovery_test() -> eyre::Result<()> {
 
     Ok(())
 }
+
+#[actix_web::test]
+/// post invalid commitment txs where tx id has been tampered with
+/// expect invalid txs to fail when sent directly to the mempool
+async fn commitment_tx_signature_validation_on_ingress_test() -> eyre::Result<()> {
+    let seconds_to_wait = 10;
+
+    initialize_tracing();
+
+    let mut genesis_config = NodeConfig::testnet();
+
+    let signer = genesis_config.new_random_signer();
+    genesis_config.fund_genesis_accounts(vec![&signer]);
+
+    let genesis_node = IrysNodeTest::new_genesis(genesis_config.clone())
+        .start()
+        .await;
+
+    //
+    // Test case 1: Stake commitment txs
+    //
+
+    // create valid and invalid stake commitment tx
+    let stake_tx = new_stake_tx(&H256::zero(), &signer, &genesis_config.consensus_config());
+    let mut stake_tx_invalid = stake_tx.clone();
+    let mut bytes = stake_tx_invalid.id.to_fixed_bytes();
+    bytes[0] ^= 0x01;
+    stake_tx_invalid.id = H256::from(bytes);
+
+    // ingest invalid stake tx directly to the mempool
+    match genesis_node
+        .ingest_commitment_tx(stake_tx_invalid.clone())
+        .await
+    {
+        Err(AddTxError::TxIngress(TxIngressError::InvalidSignature)) => {
+            // it failed to ingress, as expected!
+        }
+        Err(e) => {
+            panic!("Expected InvalidSignature but got: {:?}", e);
+        }
+        Ok(_) => {
+            panic!("Expected failure but got success");
+        }
+    }
```

Comment:
```suggestion
   let err = ...
        .await.expest_err("expected failure");   
    assert_eq!(err, AddTxError::TxIngress(TxIngressError::InvalidSignature);
```
you can make the test shorter & easier to read, same applies to other asserts